solr/contrib/ltr/src/test/org/apache/solr/ltr/ranking/TestSelectiveWeightCreation.java

```diff
@@ -257,4 +259,89 @@ public void testSelectiveWeightsRequestFeaturesFromDifferentStore() throws Excep
     assertJQ("/query" + query.toQueryString(), "/response/docs/[0]/fv=='popularity:3.0;originalScore:1.0'"); // extract all features from fstore4
   }
   
-}
\ No newline at end of file
+  @Test
+  public void testModelQueryParallelWeightCreation() throws IOException, ModelException {
+    final Directory dir = newDirectory();
+    final RandomIndexWriter w = new RandomIndexWriter(random(), dir);
+
+    Document doc = new Document();
+    doc.add(newStringField("id", "0", Field.Store.YES));
+    doc.add(newTextField("field", "wizard the the the the the oz",
+        Field.Store.NO));
+    doc.add(new FloatDocValuesField("final-score", 1.0f));
```

Comment:
All the document setup could be added into the before test/class function, so other test functions can sill use it.